projects/uart/README.md

```diff
@@ -0,0 +1,233 @@
+| `0xDD`      | ESC_ESC       | Transposed Frame Escape |
+
+#### [REQ_1030] USB custom frame
+
+The product **must** use a custom frame protocol. The frames are composed of:
+
+<img src="img/data_frame.jpg" alt="Data frame" title="Data frame">
+
+- a 16 bits request code
+- Data
+- a 16 bits CRC
+
+#### [REQ_1040] Custom frame CRC
+
+The CRC used in USB custom frame **must** be `crc-ccit-false`. It can be implemented in two different ways. By using the predefined:
+```
+crc16 = crcmod.predefined.mkCrcFun("crc-ccitt-false")
+```
+Or by using the crcmod method:
+```
+crc16 = crcmod.mkCrcFun(0x11021, rev=False, initCrc=0xFFFF, xorOut=0x0000)
+```
+
+#### [REQ_1050] Custom frame transfer machanism
+
+There **must** be two possible transfer mechanisms. For each, there can be only one request at a time before receiving an answer.
+- Standard request: the transfer is initiated by the host and wait for an answer from the host adapter
+
+<img src="img/standard_request.jpg" alt="Data frame" title="Data frame">
+
+- Notification: the transfer is initiated by the host adapter and wait for an answer from the host
+
+<img src="img/notification.jpg" alt="Data frame" title="Data frame">
+
+<br/>
+
+### SECTION 2: Generic Requests
+
+***Generic requests***
+
+| Code        | Function  |
+| ----------- | --------- |
+| `0x0000`    | Ping      |
+| `0x0001`    | ItfType   |
+| `0x0002`    | Version   |
+| `0x0003`    | IdGet     |
+
+***Shared answers***
+
+| Code        | Function      |
+| ----------- | ------------- |
+| `0xFFFF`    | Good          |
+| `0xFFFE`    | ErrGeneric    |
+| `0xFFFD`    | ErrCRC        |
+| `0xFFFC`    | ErrUnknownCode|
+| `0xFFFB`    | ErrInvalidArgs|
+| `0xFFFA`    | ErrBusy       |
+
+***Generic answers***
+
+| Code        | Function      |
+| ----------- | ------------- |
+| `0xFEFF`    | VersionResp   |
+| `0xFEFE`    | ItfTypeResp   |
+| `0xFEFD`    | IdResp        |
+
+#### [REQ_2000] Ping
+
+The product **must** answer when the ping request `0x0000` is received with the good answer `0xFFFF`.
+
+#### [REQ_2010] Interface type
+
+The product **must** answer when the ItfType request `0x0001` is received with the ItfTypeResp answer `0xFEFE`. The data returned must be `"picoha-uart"`.
+
+#### [REQ_2020] Version
+
+The product **must** answer when the Version request `0x0002` is received with the VersionResp answer `0xFEFF`. The data returned must be version of the firmware loaded on the product.
+
+#### [REQ_2030] ID
+
+The product **must** answer when the IdGet request `0x0003` is received with the IdResp answer `0xFEFD`. The data returned must be unique board ID of the product.
+
+#### [REQ_2040] Good
+
+The product **must** answer the code `0xFFFF` when no error has been encountered.
+
+#### [REQ_2050] Generic error
+
+The product **must** answer the code `0xFFFE` when a generic error is encountered.
+
+#### [REQ_2060] CRC error
+
+The product **must** answer the code `0xFFFD` when a CRC error is encountered.
```

Comment:
The product **must** answer the code `0xFFFD` when the CRC of a received request is invalid.